03-rnaseq/differential-expression_rnaseq_01.Rmd

```diff
@@ -0,0 +1,417 @@
+```{r}
+metadata <- metadata %>%
+  # the last bit of the title, separated by "-" contains the mutation
+  # information that we want to extract
+  dplyr::mutate(asxl_mutation_status = stringr::word(refinebio_title,
+    -1,
+    sep = "-"
+  )) %>%
+  # now let's make this simpler
+  dplyr::mutate(asxl_mutation_status = dplyr::case_when(
+    grepl("ASXL1|ASXL2", asxl_mutation_status) ~ "asxl_mutation",
+    grepl("ASXLwt", asxl_mutation_status) ~ "no_mutation"
+  ))
+```
+
+Let's take a look at `metadata_df` to see if this worked.
+
+```{r}
+# looking at the first 6 rows of the metadata_df and only at the columns that
+# contain the title and the mutation status we extracted from the title
+head(dplyr::select(metadata, refinebio_title, asxl_mutation_status))
+```
+
+Before we set up our model in the next step, we want to check if our modeling variable is set correctly. 
+We always want our "control" to to be set as the first level in the variable we provide as our experimental variable. 
+
+```{r}
+str(metadata$asxl_mutation_status)
+```
+
+Currently, `asxl_mutation_status` is a character.
+To make sure it is set how we want, let's mutate it to a factor so we can explicitly set the levels. 
+
+```{r}
+# Make asxl_mutation_status a factor and set the levels appropriately
+metadata <- metadata %>%
+  dplyr::mutate(
+    # Here we will set up the factor aspect of our new variable.
+    asxl_mutation_status = factor(asxl_mutation_status, levels = c("no_mutation", "asxl_mutation"))
+  )
+```
+
+Let's double check if the levels are what we want using the `levels()` function. 
+
+```{r}
+levels(metadata$asxl_mutation_status)
+```
+
+Yes! `no_mutation` is the first level as we want it to be. We're all set and ready to move on to making our `DESeq2Dataset` object. 
+
+## Create a DESeq2Dataset
+
+We will be using the `DESeq2` package for normalization and differential expression testing, which requires us to format our data into a `DESeqDataSet` object.
+First we need to prep our gene expression data frame so it's in the format that is compatible with the `DESeqDataSetFromMatrix()` function in the next step. 
+
+```{r}
+# We are making our data frame into a matrix and rounding the numbers
+gene_matrix <- round(as.matrix(df))
+```
+
+Now we need to create `DESeqDataSet` from our expression dataset. 
+We use the `asxl_mutation_status` variable we created in the design formula because that will allow us
+to model the presence/absence of _ASXL1/2_ mutation.
+
+```{r}
+ddset <- DESeqDataSetFromMatrix(
+  countData = gene_matrix,
+  colData = metadata,
+  design = ~asxl_mutation_status
+)
+```
+
+## Define a minimum counts cutoff
+
+We want to filter out the genes that have not been expressed or that have low expression counts because we want to remove any possible noise from our data before we normalize the data and perform differential expression. 
+We are going to do some pre-filtering to keep only genes with 10 or more reads in total across the samples.
```

Comment:
My understanding is that this step, for the analyses that is relevant to this notebook, is most helpful for saving on resources -- e.g., why included these genes as part of the object in memory if you will not be able to (powered to) call them as differentially expressed?